app/move/controllers/update/base.js

```diff
@@ -0,0 +1,109 @@
+const { get, keys } = require('lodash')
+
+const personService = require('../../../../common/services/person')
+const CreateBaseController = require('../create/base')
+
+class UpdateBaseController extends CreateBaseController {
+  middlewareLocals() {
+    super.middlewareLocals()
+    this.use(this.setNextStep)
+    this.use(this.setInitialStep)
+  }
+
+  _setModels(req) {
+    const res = req.res
+    req.models.move = res.locals.move
+    req.models.person = req.models.move.person
+  }
+
+  getBaseUrl(req) {
+    const moveId = req.getMoveId()
+    return `/move/${moveId}`
+  }
+
+  setCancelUrl(req, res, next) {
+    res.locals.cancelUrl = this.getBaseUrl(req)
+    next()
+  }
+
+  setNextStep(req, res, next) {
+    const nextUrl = this.getBaseUrl(req)
+    req.form.options.next = nextUrl
+    next()
+  }
+
+  setInitialStep(req, res, next) {
```

Comment:
What's the purpose of this? Is it a form-wizard thing or a thing we've added to support edit?

Reply:
Scenario:
- User goes to move A details
- User edit personal details and attempt to delete the name
- User sees error saying name can't be empty
- User goes move B details (by way of dashboard)
- User sees error again

As the user didn't exit through the gift shop, we haven't been able to delete the session.

If we set `reset` and/or `resetJourney` on these steps then the session data is dropped if there is an error resulting in the user only seeing empty values (or the original data if that's been fetched). Nor can we rely on the `checkJourney` prop because these steps are their own entry points.

Maybe the `journeyName` could be updated so as to be different for each move, but that seems a bit iffy (and I'm not even sure it would work). Even if that was so, it would mean directly going to the page (ie. where moves A and B are the same) would still trigger the error which I think is the wrong UX

Additionally, `req.initialStep` enables not reloading the move's data so it's not without utility.